lints.rs

```diff
@@ -13,36 +13,45 @@
     clippy::exit,
     clippy::explicit_into_iter_loop,
     clippy::filter_map_next,
+    clippy::float_cmp_const,
     clippy::fn_params_excessive_bools,
+    clippy::get_unwrap,
```

Comment:
`get_unwrap` is also one of those that we have tested but didn't catch any issues in our code, so been lower priority for us. but could be enabled

Reply:
Tested it on our codebase again and I'm a bit torn, actually prefer the explicit `unwrap` for temporary code so one sees that one needs to clean up the unwraps with proper error handling, instead of the more hidden indexing. 

We had a case like this for example:

```
    pub fn bounding_box(&self, node: NodeId) -> BoundingBox {
        let node = self.nodes.get(&node).unwrap();
```

Which was pretty clear that that unwrap should be replaced with some error handling and like `Result` to return, or do an explicit `.expect("Invalid node ID") // node ID needs to have been validated before` or something similar

Reply:
In hindsight, yeah that sounds like something we don't want to enable either. Rather we need a lint the other way around to never use `unwrap` but always `expect` (can be achieved by allowing `expect_used` and disallowing `unwrap_used`), and perhaps disallow direct indexing entirely ([`clippy::indexing_slicing`](https://rust-lang.github.io/rust-clippy/master/#indexing_slicing)). But then we have a lot of indexing/slicing cases where we "know" we are in bounds. Some cases could be replaced by `chunks`/`windows` though.

It doesn't seem like `const_err` is properly adhered to either; we're mapping a bunch of winit keys to imgui-rs' `Io` struct with a [constant-sized `key_map`](https://github.com/imgui-rs/imgui-rs/blob/22eac761d31aff0446f2ce6795e344bd49c74696/imgui/src/io.rs#L157) yet get a warning that every access could potentially be out of bounds :thinking: